libs/MeshKernel/include/MeshKernel/Polygons.hpp

```diff
@@ -63,6 +67,12 @@ namespace meshkernel
         /// @return The new offset polygon
         [[nodiscard]] Polygons OffsetCopy(double distance, bool innerAndOuter) const;
 
+        /// @brief Snap the polygon to the land boundary
+        ///
+        /// The polygon points are snapped to the closest point on the land boundary.
+        /// @param[in] landBoundary The land boundary to which the polygon should be snapped.
+        void SnapToLandBoundary(const LandBoundary& landBoundary);
```

Comment:
I think we also need to select the polygon indices for selecting the part of the polygon that will be snapped

Reply:
Added start and end index